IBM/carbon-addons-iot-react
# 1176 single row edit mode

Closes #1176

**Summary**
This allows users to quickly, and relatively simply edit individual records in a table, without needing to flip the whole thing into a bulk edit mode. Its mostly a nice UX affordance to reduce errors, but also likely helps performance where entire tables cannot be flipped into edit, then submitted.

Table interaction is suspended while in single row edit mode and since the toolbar is still showing it had to be completely disabled, including any active filters and column reorder that were showing. Simply hiding them like we do for the full table edit mode would not be a good solution here since the rows would jump vertically, moving the editable row away from the mouse pointer. That made this PR grow a bit in size and complexity.

**Change List (commits, features, bugs, etc)**
- public `isEdit` prop on RowAction objects
- public `isEditMode` prop on RowActionsState objects
- public `view.table.singleRowEditButtons` prop for app defined save/cancel buttons
- public `options.hasSingleRowEdit` prop added. Needed to set consistent larger width of action row column
- updated stateful table to handle the edit row action
- updated stories with examples and texts
- updated unit tests
- added disable option to the TableToolBar buttons


**Acceptance Test (how to verify the PR)**
1. Go to table story [watson-iot-table--basic-table-with-full-rowedit-example](https://deploy-preview-1204--carbon-addons-iot-react.netlify.app/?path=/story/watson-iot-table--basic-table-with-full-rowedit-example)
2. Hover a row and click the row action button
3. Select edit. Make sure the table is disabled except for the row that you want to edit.
4. Try different scenarios of editing/save/cancel/undo
5. Uncheck the hasSingleRowEdit and make sure the row action column gets smaller

6. Go to table story [on-iot-table--stateful-example-with-expansion-and-column-resize](https://deploy-preview-1204--carbon-addons-iot-react.netlify.app/?path=/story/watson-iot-table--stateful-example-with-expansion-and-column-resize) which contains lots of functionality, including a single row edit action.
7. Hover a row and click the row action button
8. Select edit. Make sure all the functionality of the table is disabled (the actual row edit is not implemented here)
9. Try the same story and scenario again but have different toolbars active. Make sure they are all disabled.
